generator/mavgen_java.py

```diff
@@ -93,9 +101,29 @@ def generate_CRC(directory, xml):
     *
     * @param msgid
     *            The message id number
+    * @return crc_extra
+    *            The crc_extra value for the given message
     */
-    public void finish_checksum(int msgid) {
-        update_checksum(MAVLINK_MESSAGE_CRCS[msgid]);
+    public int finish_checksum(int msgid) {
+        int crc = 0;
+        if (msgid == 0) {
+            /* special handling to high frequency heartbeat message */
+            crc = MAVLINK_MESSAGE_CRCS[0];
+        } else {
+            int low = 0, high = MAVLINK_MESSAGE_IDS.length - 1;
```

Comment:
Why is this rolling it's own binary search? You have a standard library which provides a fast and stable solution to this for you. https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#binarySearch(int[],%20int)